shell/server/mail.js

```diff
@@ -377,18 +378,23 @@ HackSessionContextImpl.prototype.getPublicId = function() {
   }).bind(this));
 };
 
+var findRequestMethod = function (url) {
+  if (url.indexOf('https://') === 0) {
```

Comment:
Nitpick: It's better to write: `url.lastIndexOf('https://', 0)`

Otherwise when the url doesn't start with `https://` you'll needlessly search the whole string.

Also, even more nitpicky: can we try to stick to double-quotes? I know, hard to change habits.